imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java

```diff
@@ -775,6 +806,52 @@ public void contentCompleted() {
         }
     }
 
+    @Override
+    public void playAdNow(String adTagUrl) {
+        if (isAdvertisingConfigured) {
+            isAdvertisingConfigLoading = !TextUtils.isEmpty(adTagUrl);
+            requestAdsFromIMA(null, adTagUrl);
+        }
+    }
+
+    @Override
+    public void setAdvertisingConfig(boolean isConfigured, IMAEventsListener imaEventsListener) {
+        isAdvertisingConfigured = isConfigured;
+        this.imaEventsListener = imaEventsListener;
```

Comment:
@GouravSna  where do you remove this listener when done?

Reply:
done